engine/crates/parser-sdl/src/validations.rs

```diff
@@ -462,14 +462,28 @@ fn types_are_compatible(mut actual: &Type, mut expected: &Type, registry: &Regis
                 if actual_name == expected_name {
                     return true;
                 }
-                match (registry.lookup(actual_name), registry.lookup(expected_name)) {
-                    (Ok(MetaType::Scalar(_)), Ok(MetaType::Scalar(_))) => {
+                match registry
+                    .lookup(actual_name)
+                    .ok()
+                    .zip(registry.lookup(expected_name).ok())
+                {
+                    Some((MetaType::Scalar(_), MetaType::Scalar(_))) => {
                         // If the names don't match but both sides are scalars we'll say they're compatible.
                         // This maybe isn't strictly correct but a bit of flexibility around
                         // passsing strings to ID fields and handling mismatched custom scalars
                         // seems sensible.
                         return true;
                     }
+                    Some(
+                        (MetaType::InputObject(_), MetaType::Object(_))
+                        | (MetaType::Object(_), MetaType::InputObject(_)),
```

Comment:
Might need to do something similar around enums here to handle trip advisors case :|